libraries/AP_Relay/AP_Relay.cpp

```diff
@@ -140,6 +141,13 @@ void AP_Relay::set(const uint8_t instance, const bool value)
     if (_pin[instance] == -1) {
         return;
     }
+    _pin_states = value?_pin_states | (1<< instance): _pin_states & ~(1<< instance);
+    AP::logger().Write("RELY", "TimeUS,State", "QB",
```

Comment:
Possibly need to rate limit this in some way in case the relays are being switched very fast.

Reply:
I dont know of any way to toggle a relay fast....actuation methods are:
MAVLink message
Parachute code
Aux Switch Function (rc/button)
Camera Trigger
Mission command
Motor reverse

you could mess up code to do that, but you have bigger problems than getting a lot of log messages....

Reply:
so would I need to put a timer based off hal_millis to prevent that...just have it update once per second max?

Reply:
added